lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/PointToFieldFunctionalChaosAlgorithm.cxx

```diff
@@ -0,0 +1,347 @@
+    klResultCollection[i] = algo.getResult();
+    LOGINFO(OSS() << "block #" << i << " indices=" << blockIndices_[i] << " ev size=" << algo.getResult().getEigenvalues().getDimension());
+  }
+
+  if (recompress_)
+  {
+    const Scalar cumulatedVariance = eigenValues.norm1();
+    std::sort(eigenValues.begin(), eigenValues.end(), [](const Scalar a, const Scalar b) { return a > b; });
+    const UnsignedInteger nbModesMax = std::min(nbModes_, eigenValues.getSize());
+    // Find the cut-off in the eigenvalues
+    UnsignedInteger K = 0;
+    Scalar selectedVariance = 0.0; // sum of eigenvalues selected after cut-off is applied
+    do
+    {
+      selectedVariance += eigenValues[K];
+      ++ K;
+    }
+    while ((K < nbModesMax-1) && (selectedVariance < (1.0 - threshold_) * cumulatedVariance));
+    LOGINFO(OSS() << "Selected " << K << " eigenvalues out of " << eigenValues.getSize() << " computed");
+
+    const Scalar lambdaCut = eigenValues[K];
+    for (UnsignedInteger i = 0; i < blockIndices_.getSize(); ++ i)
+    {
+      UnsignedInteger Ki = 1;// retain at least 1 mode
+      Point eigenValuesI(klResultCollection[i].getEigenvalues());
+      Scalar selectedVarianceI = eigenValuesI[0];
+      const Scalar cumulatedVarianceI = eigenValuesI.norm1();
+      while ((Ki < eigenValuesI.getSize()) && (eigenValuesI[Ki] > lambdaCut))
+      {
+        selectedVarianceI += eigenValuesI[Ki];
+        ++ Ki;
+      }
+      eigenValuesI.resize(Ki);
+      Collection<Function> modesI(klResultCollection[i].getModes());
+      modesI.resize(Ki);
+      CovarianceModel covarianceI(RankMCovarianceModel(eigenValuesI, modesI));
+      ProcessSample modesAsProcessSampleI(klResultCollection[i].getModesAsProcessSample());
+      modesAsProcessSampleI.erase(Ki, modesAsProcessSampleI.getSize());
+      MatrixImplementation projectionMatrixI(*klResultCollection[i].getProjectionMatrix().getImplementation());
+      projectionMatrixI.resize(Ki, projectionMatrixI.getNbColumns());
+
+      klResultCollection[i] = KarhunenLoeveResult(covarianceI, klResultCollection[i].getThreshold(), eigenValuesI, modesI, modesAsProcessSampleI, projectionMatrixI, selectedVarianceI / cumulatedVarianceI);
+    }
+  }
+
+  // the global input projection stacks projections of each block of variables
+  Sample modesSample(size, 0);
+  Collection<PointToFieldFunction> liftingCollection(blockIndices_.getSize());
+  for (UnsignedInteger i = 0; i < blockIndices_.getSize(); ++ i)
+  {
+    const KarhunenLoeveProjection projectionI(klResultCollection[i]);
+    const KarhunenLoeveLifting liftingI(klResultCollection[i]);
+    liftingCollection[i] = liftingI;
+    const ProcessSample outputProcessSampleI(outputProcessSample.getMarginal(blockIndices_[i]));
+    const Sample modesSampleI(projectionI(outputProcessSampleI));
+    modesSample.stack(modesSampleI);
+  }
+
+  // build PCE expansion of projected modes vs output sample
+  LeastSquaresExpansion fce(inputSample_, modesSample, distribution_);
```

Comment:
is there a reason why it's explicitely LSE ? Might be Integration or LARSExpansion (one day) ?

Reply:
I double IntegrationExpansion would be efficient here as we could have many outputs
but why not LARSExpansion, when it becomes available

Reply:
I agree that LeastSquaresExpansion is, in general, better than IntegrationExpansion both in terms of speed and accuracy. But IMO the simplified version of FunctionalChaosAlgorithm should also be available through eg a key in ResourceMap, as it is quite efficient in terms of accuracy and can be fine-tuned through ResourceMap. Something like:
```
FunctionalChaosResult fceResult;
if (ResourceMap::GetAsString("PointToFieldFunctionalChaosAlgorithm-DefaultExpansion") == "LeastSquaresExpansion")
{
  LeastSquaresExpansion fce(inputSample_, modesSample, distribution_);
  fce.run();
  fceResult = fce.getResult();
}
else
{
  FunctionalChaosAlgorithm fce(inputSample_, modesSample, distribution_);
  fce.run();
  fceResult = fce.getResult();
}
```
and LARSExpansion could then be put in the second block when available.